File: 2025/day01/prob1.py
def main():
    with open("2025/day01/input.txt") as f:
        rotations = f.read().split("\n")
    position = 50
    n = 0
    for rotation in rotations:
        direction = rotation[0]
        distance = int(rotation[1:])
        if direction == 'L':
            position -= distance
        else:
            position += distance
        position %= 100
        if position == 0:
            n += 1
    print(n)


if __name__ == "__main__":
    main()

File: 2025/day01/Prob1.java
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.List;

public class Prob1 {

    public static void main(String args[]) throws Exception {
        List<String> rotations = Files.readAllLines(Path.of("2025/day01/input.txt"));
        int position = 50;
        int n = 0;
        for (String rotation: rotations) {
            char direction = rotation.charAt(0);
            int distance = Integer.parseInt(rotation.substring(1));
            if (direction == 'L') {
                position -= distance;
            } else {
                position += distance;
            }
            position %= 100;
            if (position == 0) {
                n++;
            }
        }
        System.out.println(n);
    }

}